src/graph.h

```diff
@@ -103,6 +105,12 @@ struct Node {
 
   void Dump(const char* prefix="") const;
 
+  /// Used in the inputs debug tool.
+  bool InputsChecked() const { return inputs_checked_; }
+  void MarkInputsChecked() { inputs_checked_ = true; }
+  bool UsedAsDep() const { return used_as_dep_; }
+  void MarkUsedAsDep() { used_as_dep_ = true; }
```

Comment:
Does any of this state need to be cleared when the tool is done running? I'm not sure what gets cached in ninja.